src/backend/distributed/deparser/deparse_table_stmts.c

```diff
@@ -229,6 +240,36 @@ AppendAlterTableCmdAddConstraint(StringInfo buf, Constraint *constraint)
 
 		appendStringInfoString(buf, " )");
 	}
+	else if (constraint->contype == CONSTR_CHECK)
+	{
+		LOCKMODE lockmode = AlterTableGetLockLevel(stmt->cmds);
+		Oid leftRelationId = AlterTableLookupRelation(stmt, lockmode);
+
+		/* To use deparse_expression to convert expression into a string
+		 * the expression should be in cooked form. We transform the raw expression
+		 * to cooked form.
+		 */
+		ParseState *pstate = make_parsestate(NULL);
+		Relation relation = table_open(leftRelationId, AccessShareLock);
+
+		/* Add table name to the name space in  parse state. Otherwise column names
+		 * cannot be found.
+		 */
+		AddRangeTableEntryToQueryCompat(pstate, relation);
+
+		Node *expr = transformExpr(pstate, constraint->raw_expr,
+
+								   EXPR_KIND_CHECK_CONSTRAINT);
+
+		char *relationName = get_rel_name(leftRelationId);
+		List *deparseCtx = deparse_context_for(relationName, leftRelationId);
+
+		char *exprSql = deparse_expression(expr, deparseCtx, false, false);
+
+		relation_close(relation, NoLock);
+
+		appendStringInfo(buf, " CHECK (%s)", exprSql);
```

Comment:
check constraints can be followed by `NO INHERIT`, I guess you should add that to the deparser and some test cases along with it as well.

Reply:
Per my testing, tables in an inheritance hierarchy cannot be distributed. Hence handling "NO INHERIT" is irrelevant in this codepath since no such clause will be run by the citus utility hook.

Reply:
You are right. I thought this somehow is valid for partitioned tables as well, but it's not. In that case, maybe you can now add an "if" statement (currently unreachable) or something like that.
```
if(constraint->noinherit) {
   error out
}
```